src/SocketIOHandler.ts

```diff
@@ -1,205 +1,286 @@
+            }
+            return;
         }
-        if (value) {
-          return next();
+        logger.debug("... message payload was successfully parsed.", TAG);
+
+        if (data.hasOwnProperty("metadata")) {
+            if (!data.metadata.hasOwnProperty("deviceid")) {
+                logger.debug("... received message was not successfully processed.", TAG);
+                logger.error("Received data is not a valid dojot event - has no deviceid", TAG);
+                return;
+            }
+        } else {
+            logger.debug("... received message was not successfully processed.", TAG);
+            logger.error("Received data is not a valid dojot event - has no metadata", TAG);
+            return;
+        }
+
+        logger.debug(`Will publish event to namespace ${nsp} from device ${data.metadata.deviceid}`,
+            {filename: "SocketIOHandler"});
+        this.ioServer.to(nsp).emit(data.metadata.deviceid, data);
+        this.ioServer.to(nsp).emit("all", data);
+        logger.debug("... received message was successfully processed.", TAG);
+    }
+
+    /**
+     * Callback function used to process actuator message  received from Kafka library.
+     * @param nsp SocketIO namespace to send out messages to all subscribers. These are tenants.
+     * @param error Error received from Kafka library.
+     * @param message The actuator message  received from Kafka Library
+     */
+    private handleMessageActuator(nsp: string, message: string) {
+        logger.debug("Processing actuator message just received...", TAG);
+        let data: any;
+        logger.debug("Trying to parse received actuator message payload...", TAG);
+        try {
+            data = JSON.parse(message);
+        } catch (err) {
+            if (err instanceof TypeError) {
+                logger.debug("... actuator message payload was not successfully parsed.", TAG);
+                logger.error(`Received data is not a valid event: ${message}`, TAG);
+            } else if (err instanceof SyntaxError) {
+                logger.debug("... actuator message payload was not successfully parsed.", TAG);
+                logger.error(`Failed to parse event as JSON: ${message}`, TAG);
+            }
+            return;
+        }
+        logger.debug("... actuator message payload was successfully parsed.", TAG);
+
+        if (!data.hasOwnProperty("event")) {
+            logger.debug("... received actuator message was not successfully processed.", TAG);
+            logger.error("Received data is not a valid dojot event - has no event", TAG);
+            return;
+        } else if (data.event !== "configure") {
+            logger.debug("... received actuator message was not successfully processed.", TAG);
+            logger.error("Received data is not a valid dojot event - event is not configure", TAG);
+            return;
+        }
+
+        if (!data.hasOwnProperty("meta")) {
```

Comment:
Data Manager uses a library (express-validator, I think) to check the validity of incoming JSON objects. How about give it a try in the future and see if it becomes clearer?

Reply:
Thx :)
It really sounds interesting